/*
 * LSST Data Management System
 * Copyright 2018 LSST Corporation.
 *
 * This product includes software developed by the
 * LSST Project (http://www.lsst.org/).
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the LSST License Statement and
 * the GNU General Public License along with this program.  If not,
 * see <http://www.lsstcorp.org/LegalNotices/>.
 */
/// replication.proto
/// This defines the wire-messages sent between replication master and workers.

package lsst.qserv.proto;

/// Response to any maessage
message LdrMsgReceived {
    /// 
    required uint64 originalid   = 1;
    required uint32 originalkind = 2;
    required uint32 status       = 3;
    optional string errmsg       = 4;
    required uint32 dataentries  = 5;
}

/// Register a worker with the master.    
message LdrMastWorkerAddReq {       /// Change the name to NetAddress &&&
    required string workerip     = 1;
    required uint32 workerport   = 2;
}


message WorkerRangeString {
    required string min  = 1; 
    required string max  = 2;
}


message WorkerRangeInt {
	required uint64 minint  = 1;
	required uint64 maxint  = 2; 
}


message WorkerListItem {
    required uint32 name    = 1; // worker's id number
    optional LdrMastWorkerAddReq address = 2;
    optional WorkerRangeString rangeStr = 3;
    optional WorkerRangeInt rangeInt = 4;
}


message LdrMastWorkerList {
	required uint32         workercount = 1;
    repeated WorkerListItem worker      = 2;
}



 